pkg/services/ngalert/provisioning/templates.go

```diff
@@ -53,6 +53,7 @@ func (t *TemplateService) SetTemplate(ctx context.Context, orgID int64, tmpl def
 		revision.cfg.TemplateFiles = map[string]string{}
 	}
 	revision.cfg.TemplateFiles[tmpl.Name] = tmpl.Template
+	revision.cfg.AlertmanagerConfig.Templates = append(revision.cfg.AlertmanagerConfig.Templates, tmpl.Name)
```

Comment:
This will also append each time a provisioned template is updated:

```
curl -H "Content-Type: application/json" -u admin:admin http://127.0.0.1:3000/api/v1/provisioning/templates/test -X PUT -d '{"name":"test","template":"hello1"}'
curl -H "Content-Type: application/json" -u admin:admin http://127.0.0.1:3000/api/v1/provisioning/templates/test -X PUT -d '{"name":"test","template":"hello2"}'
```
```
"templates": [
  "test",
  "test"
],
```

Reply:
I guess we might as well overwrite `cfg.AlertmanagerConfig.Templates` here with the current state of `TemplateFiles` since we'll do it during `applyConfig` anyways, WDYT?

Reply:
Yes I think so! I've noticed that sometimes deleted templates are removed from `template_files` and not `templates`, so it should fix both!

Reply:
What's the relationship between this and `cfg.AlertmanagerConfig.Templates = paths` on Line 269 of `pkg/services/ngalert/notifier/alertmanager.go`.

Reply:
This change is not technically required since we now do the same thing in `pkg/services/ngalert/notifier/alertmanager.go`, figured we might preempt future issues by just making this do both anyways. I can remove it if that's the preference.